### DISCUSSION TITLE
Controller component with array has undefined values, if there is no default entry

### DISCUSSION THREAD

**Discussion by BuckUbel** [react-hook-form/react-hook-form#5873] [Q&A]
**Describe the error**
Undefined values when using Controller component for array values without default entry

**Reproduce**

Create a form.
Add an array for form values like string[].
Add an empty array to it as the default value.
Use setValue to add entries to the formState.
If you use the controller component to render an input field with rules, you will only get undefined values.
(see codesanbox)

**Codesandbox**
https://codesandbox.io/s/react-hook-forms-control-arrays-7qx69?file=/src/App.tsx
Case 1: Add answers without Controller with empty defaultValues-array 🟢
Case 2: Add answers with Controller with empty defaultValues-array 🔴
 - the previous values are strangely deleted

Case 3: Add answers without Controller with defaultValues-entry 🟢
Case 4: Add answers with Controller with defaultValues-entry 🟢

I would also expect the behavior of case 4 in case 2. 
But why is it not like that there?

**Desktop**
OS: Windows 10
Browser: Chrome
Version: 91
react-hook-form version: 7.10.1

**Comment by bluebill1049** [comment]
can you make your codesandbox isolated to the single problem? easier for me to look into. thanks

**Reply by BuckUbel** [reply]
Sure, no problem. 

https://codesandbox.io/s/react-hook-forms-control-arrays-forked-tq8lp?file=/src/App.tsx
